### PR TITLE
Post feature

### PR DESCRIPTION
**Pull request by zilton7**
In this project we implemented: 

- Authentication, using devise gem;
- Forced the user to signup/login before creating new post;
- Made post author visible only to signed in users.

### CODE REVIEW

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @zilton7 and @marijanbrvar    ! :wave: :wave:

#### You have done well so far, your project works as expected.  :clap: However, there are changes required for you to make to be approved for this project. Kindly work on them and submit them for another code review.
- - -

### **Recommended**
- [x]  User can sign_out ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

* [x] The application has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))

- [x] The root page should have links to create new posts.
- [x] Adjust the layout of the post(index page) to show in order of name, text and author. It can be added like a table. It currently shows as a list, making it seem like a list of different posts instead of just one.
- [x] only user routes should be exposed. routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

##### PostsController
###### `create` action

* [x] it makes sure that the user_id for post is the one from the signed_in user ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_create_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

```ruby
# the best way to do this
@post = current_user.posts.build(post_params)

#this could also work
@post = Post.new(post_params)
@post.user_id = current_user.id

# IMPORTANT: user_id cannot be passed in params (security)
```


__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,8 +9,15 @@
...
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
-
   <body>
+  <br/>
+  <% if signed_in? %>
+    <%= link_to 'sign out', destroy_user_session_path, method: :delete  %>
+  <% else %>
+    <%= link_to 'sign in', new_user_session_path %> / <%= link_to 'sign up', new_user_registration_path %>
+  <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:18]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @zilton7 and @marijanbrvar, Great job! 👏  Both of you have done well on this project.
✔️ Your ReadMe file looks good 👍 
✔️  Great codebase

Let's make a few changes, and we will be done.

- [x] Add `yarn` to your getting started instructions, perfect after the `bundle install`. I couldn't run the project without running `yarn`
- [x] I should not be able to create an empty post, kindly add validation to your post title and body.

### Consider the requested changes below as OPTIONAL, since the previous TSE may have missed it. If you can try to implement it before submitting another code review.

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

- [ ] Your commit messages should be capitalized. You can watch this [video](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6) on how to edit your commit messages using git rebase
 See [Git Commit Best Practices](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) for more insights

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by oloomoses — APPROVED** [review]
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

## Optional
- Kindly consider implementing the optional changes requested by the previous reviewer before moving forward.

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Pull request merged**